From 4fcab9a88bf036d2a5b38e3655099f79b5faf695 Mon Sep 17 00:00:00 2001
From: Mark Allen Evans <president@worldvaporexpo.com>
Date: Wed, 11 Apr 2018 20:12:41 -0400
Subject: [PATCH] Update blockchain.cpp

---
 src/cryptonote_core/blockchain.cpp | 10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

diff --git a/src/cryptonote_core/blockchain.cpp b/src/cryptonote_core/blockchain.cpp
index ae26155a..d9944164 100644
--- a/src/cryptonote_core/blockchain.cpp
+++ b/src/cryptonote_core/blockchain.cpp
@@ -58,7 +58,7 @@
 
 #undef MONERO_DEFAULT_LOG_CATEGORY
 #define MONERO_DEFAULT_LOG_CATEGORY "blockchain"
-
+#define MAINNET_HARDFORK_V2_HEIGHT  ((uint64_t)(179721))
 #define FIND_BLOCKCHAIN_SUPPLEMENT_MAX_SIZE (100*1024*1024) // 100 MB
 
 //#include "serialization/json_archive.h"
@@ -89,13 +89,13 @@ static const struct {
 } mainnet_hard_forks[] = {
   // version 1 from the start of the blockchain
   { 1, 1, 0, 1341378000 },
-  // { 2, 100, 0, 1535913139 },
+  { 2, MAINNET_HARDFORK_V2_HEIGHT, 0, 1523491942 },
   // { 3, 110, 0, 1535913139 },
   // { 4, 125, 0, 1535913139 },
   // { 5, 150, 0, 1535913139 },
   // { 6, 200, 0, 1535913139 },  
 };
-static const uint64_t mainnet_hard_fork_version_1_till = 200000;
+static const uint64_t mainnet_hard_fork_version_1_till = MAINNET_HARDFORK_V2_HEIGHT-1;
 
 static const struct {
   uint8_t version;
@@ -703,6 +703,10 @@ difficulty_type Blockchain::get_difficulty_for_next_block()
   std::vector<uint64_t> timestamps;
   std::vector<difficulty_type> difficulties;
   auto height = m_db->height();
+  // Reset network hashrate to 2.0 MHz when hardfork v3 comes
+  if (!m_testnet && (uint64_t)height >= MAINNET_HARDFORK_V2_HEIGHT+1 && (uint64_t)height <= MAINNET_HARDFORK_V2_HEIGHT + (uint64_t)DIFFICULTY_BLOCKS_COUNT){
+    return (difficulty_type) 480000000;
+  }
   // ND: Speedup
   // 1. Keep a list of the last 735 (or less) blocks that is used to compute difficulty,
   //    then when the next block difficulty is queried, push the latest height data and
